src/cascadia/TerminalApp/ProfileConversionTraits.cpp

```diff
@@ -0,0 +1,178 @@
+        pair_type{ ImageAlignmentBottomLeft, std::make_tuple(HorizontalAlignment::Left, VerticalAlignment::Bottom) },
+        pair_type{ ImageAlignmentLeft, std::make_tuple(HorizontalAlignment::Left, VerticalAlignment::Center) },
+        pair_type{ ImageAlignmentTopRight, std::make_tuple(HorizontalAlignment::Right, VerticalAlignment::Top) },
+        pair_type{ ImageAlignmentBottomRight, std::make_tuple(HorizontalAlignment::Right, VerticalAlignment::Bottom) },
+        pair_type{ ImageAlignmentRight, std::make_tuple(HorizontalAlignment::Right, VerticalAlignment::Center) },
+        pair_type{ ImageAlignmentTop, std::make_tuple(HorizontalAlignment::Center, VerticalAlignment::Top) },
+        pair_type{ ImageAlignmentBottom, std::make_tuple(HorizontalAlignment::Center, VerticalAlignment::Bottom) }
+    };
+};
+
+JSON_ENUM_MAPPER(TextAntialiasingMode)
+{
+    // Possible values for TextAntialiasingMode
+    static constexpr std::string_view AntialiasingModeGrayscale{ "grayscale" };
+    static constexpr std::string_view AntialiasingModeCleartype{ "cleartype" };
+    static constexpr std::string_view AntialiasingModeAliased{ "aliased" };
+
+    static constexpr std::array<pair_type, 3> mappings = {
+        pair_type{ AntialiasingModeGrayscale, TextAntialiasingMode::Grayscale },
+        pair_type{ AntialiasingModeCleartype, TextAntialiasingMode::Cleartype },
+        pair_type{ AntialiasingModeAliased, TextAntialiasingMode::Aliased }
+    };
+};
+
+// Method Description:
+// - Helper function for converting a user-specified closeOnExit value to its corresponding enum
+// Arguments:
+// - The value from the profiles.json file
+// Return Value:
+// - The corresponding enum value which maps to the string provided by the user
+JSON_ENUM_MAPPER(CloseOnExitMode)
+{
+    // Possible values for closeOnExit
+    static constexpr std::string_view CloseOnExitAlways{ "always" };
+    static constexpr std::string_view CloseOnExitGraceful{ "graceful" };
+    static constexpr std::string_view CloseOnExitNever{ "never" };
+
+    JSON_MAPPINGS(3) = {
+        pair_type{ CloseOnExitAlways, CloseOnExitMode::Always },
+        pair_type{ CloseOnExitGraceful, CloseOnExitMode::Graceful },
+        pair_type{ CloseOnExitNever, CloseOnExitMode::Never },
+    };
+
+    // Override mapping parser to add boolean parsing
+    CloseOnExitMode FromJson(const Json::Value& json)
+    {
+        if (json.isBool())
+        {
+            return json.asBool() ? CloseOnExitMode::Graceful : CloseOnExitMode::Never;
+        }
+        return EnumMapper::FromJson(json);
+    }
+
+    bool CanConvert(const Json::Value& json)
+    {
+        return EnumMapper::CanConvert(json) || json.isBool();
+    }
+};
+
+template<>
+struct JsonUtils::ConversionTrait<winrt::Windows::UI::Text::FontWeight> : public JsonUtils::EnumMapper<unsigned int, JsonUtils::ConversionTrait<winrt::Windows::UI::Text::FontWeight>>

```

Comment:
dustin: explain why you didn't use the macro here

it is because ... oh, I am not using an enum type. okay. i'm using `unsigned int` _inside the mappings_ and returning a struct.